codecov_cli/services/staticanalysis/analyzers/python/node_wrappers.py

```diff
@@ -10,6 +10,42 @@ def visit(self, node):
         for c in node.children:
             self.visit(c)
 
+    def _is_function_docstring(self, node):
+        """Skips docstrings for funtions, such as this one.
+        Pytest doesn't include them in the report, so I don't think we should either,
+        at least for now.
+        """
+        # Docstrings for a module are OK - they show up in pytest result
+        # Docstrings for a class are OK - they show up in pytest result
+        # Docstrings for functions are NOT OK - they DONT show up in pytest result
+        # Check if it's docstring
+        has_single_child = len(node.children) == 1
+        only_child_is_string = node.children[0].type == "string"
+        # Check if is the first line of a function
+        parent_is_block = node.parent.type == "block"
+        first_exp_in_block = node.prev_named_sibling is None
+        is_in_function_context = (
+            node.parent.parent.type == "function_definition"
+            if parent_is_block and node.parent is not None
+            else False
+        )
+
+        return (
+            has_single_child
+            and only_child_is_string
+            and parent_is_block
+            and first_exp_in_block
+            and is_in_function_context
+        )
+
+    def _get_previous_sibling_that_is_not_comment_not_func_docstring(self, node):
```

Comment:
is a function docstring a "named" node? what does "named" mean?

Reply:
So a "named" node (as far as I understand) is a node that has a type. Idk what nodes don't have a type tho.

The function docstring _is_ named, but its type is not "docstring", it's "expression".
So the function `_is_function_docstring` looks at other characteristics of that expression to determine if it's a function docstring or not.